### PR TITLE
Telemetry

### PR DESCRIPTION
**Pull request by iunanua**
### What does this PR do?

Allows `Rewriter` to be configured to collect some instrumentation related metrics.
The metrics we are interested are the the number of propagation methods replaced by the `Rewriter` and details about that methods.

NOTE: The signature of `Rewriter.rewrite` method has changed and now returns an object with the content and telemetry metrics if any instead of the content of the rewrited file.

### Motivation

This PR is part of an effort to provide the IAST with startup and runtime metrics.

### Checklist

- [ ] The CHANGELOG.md has been updated
- [ ] Unit tests have been updated and pass
- [ ] If known, an appropriate milestone has been selected

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -5,17 +5,20 @@
...
 }
 export interface RewriterConfig {
   chainSourceMap?: boolean
   comments?: boolean
   localVarPrefix?: string
-  csiMethods?: Array<CsiMethod> | undefined | null
+  csiMethods?: Array<CsiMethod>
+}
+export interface ResultWithoutMetrics {
+  content: string
```
Comment (uurien) [inline index.d.ts:18]: And the `metrics`?

File: src/lib_wasm.rs [hunk 2]
```diff
@@ -4,11 +4,16 @@
 **/
 extern crate base64;
 
+use std::collections::HashMap;
+
 use crate::{
-    rewriter::{print_js, rewrite_js},
+    rewriter::{print_js, rewrite_js, Config},
+    telemetry::{Telemetry, TelemetryVerbosity},
```
Comment (uurien) [inline src/lib_wasm.rs:11]: Why do you need `Telemetry` here but not in `src/lib_napi.rs`?

File: index.d.ts (hunk 1 above)
Comment (iunanua) [reply index.d.ts:18]: NAPI interface is not used anymore so I skipped the metrics implementation

File: src/lib_wasm.rs (hunk 2 above)
Comment (iunanua) [reply src/lib_wasm.rs:11]: The same as above

**Review by hoolioh — APPROVED** [review]

**Pull request merged**